### PR TITLE
P2P sync update

### PR DESCRIPTION
**Pull request by IronGauntlets**

### CODE REVIEW

File: core/state_update.go [hunk 1]
```diff
@@ -35,6 +35,21 @@ func EmptyStateDiff() *StateDiff {
...
+	var length uint64
+
+	for _, storageDiff := range d.StorageDiffs {
+		length += uint64(len(storageDiff))
+	}
+	length += uint64(len(d.Nonces))
+	length += uint64(len(d.DeployedContracts))
+	length += uint64(len(d.DeclaredV0Classes))
+	length += uint64(len(d.DeclaredV1Classes))
+	length += uint64(len(d.ReplacedClasses))
```
Comment (kirugan) [inline core/state_update.go:48]: maybe we should make `length` var as int and cast it into `uint64` at the bottom?
I don't think that in practice sum of all these fields will reach 2^32 (max value of int)

File: adapters/core2p2p/block.go [hunk 2]
```diff
@@ -44,12 +46,15 @@ func AdaptHeader(header *core.Header, commitments *core.BlockCommitments) *spec.
 			Root:    AdaptHash(commitments.EventCommitment),
 		},
 		// todo fill receipts
```
Comment (kirugan) [inline adapters/core2p2p/block.go:48]: do we need to replace this comment?

File: adapters/core2p2p/block.go [hunk 3]
```diff
@@ -44,12 +46,15 @@ func AdaptHeader(header *core.Header, commitments *core.BlockCommitments) *spec.
...
-		Receipts:        nil,
+		Receipts:        AdaptHash(&felt.Zero),
 		ProtocolVersion: header.ProtocolVersion,
 		GasPriceFri:     AdaptUint128(header.GasPrice),
 		Signatures:      utils.Map(header.Signatures, AdaptSignature),
 		// todo(kirill) set these fields
```
Comment (kirugan) [inline adapters/core2p2p/block.go:53]: and remove this one ?

File: adapters/core2p2p/block.go (hunk 2 above)
Comment (IronGauntlets) [reply adapters/core2p2p/block.go]: Replaced with a more specific comment

File: adapters/core2p2p/block.go (hunk 3 above)
Comment (IronGauntlets) [reply adapters/core2p2p/block.go]: Done

File: core/state_update.go (hunk 1 above)
Comment (IronGauntlets) [reply core/state_update.go]: Done

**Review by kirugan — APPROVED** [review]

**Review by rianhughes — APPROVED** [review]

**Pull request merged**